libs/pika/async_mpi/include/pika/async_mpi/mpi_polling.hpp

```diff
@@ -115,57 +117,81 @@ namespace pika::mpi::experimental {
             /// from the polling thread onto a new one
             completion_inline = 0x04,
 
-            /// this bit enables the use of a high priority task flag when a
-            /// completion is handled so that the continuation is executed quickly
-            /// if transferred to a new task, or resumed from sleep.
+            /// this bit enables the use of a high priority task flag
+            /// 1) requests are boosted to high priority if they are passed the the mpi-pool
+            ///    to ensure they execute before other polling tasks (reduce latency)
+            /// 2) completions are boosted to high priority when sent to the main thread pool
+            ///    so that the continuation is executed as quickly as possible
             high_priority = 0x08,
 
-            /// 2 bits control the handler method,
-            method_mask = 0x30,
-
-            /// the individual methods that are supported for dispatching continuations
-            yield_while = 0x00,
-            suspend_resume = 0x10,
-            new_task = 0x20,
-            continuation = 0x30,
+            /// 3 bits control the handler method,
+            method_mask = 0x70,
```

Comment:
Optional refactoring: use binary literals instead, which in my opinion would be easier to read:
```suggestion
            method_mask = 0b0111'0000,
```
Likewise for the other flags.

However, related to this I think at the very least for the user this bitmask interface is terrible. I think separating the different options into separate enums in the environment variables at minimum would be good, if not internally. Is the optimal packing of these flags important for performance? This is not a blocker, and I'd be happy to do this refactoring separately.

Reply:
I was actually in the process of changing to a binary bitmask, so I'm in agreement here. I also agree that other flags would be useful, but I'd rather do that in another PR. It's just too convenient to have an integer in testing loops to scan over all of them at the moment.

Reply:
Understandable. Do you want to at least do the hex to binary literal change still in this PR, and we can discuss how best to deal with the flags separately?